app/Http/Controllers/Auth/LoginController.php

```diff
@@ -37,4 +49,139 @@ public function __construct()
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector|\Illuminate\View\View*
+     */
+    public function showLoginForm()
+    {
+        $attemptEndTime = session()->get('attempt_end_time');
+        $now = Carbon::now();
+
+        if(isset($attemptEndTime) && $attemptEndTime->diffInMinutes($now)<= 60){
+            $timeLeft = $attemptEndTime->diffInMinutes($now);
+            $timeLeft1 = 60 - $timeLeft;
+
+            return redirect('/')->withErrors(["Try again after: " . $timeLeft1 . ' minutes']);
+        }
+        else{
+
+            session()->forget('attempt_end_time');
+            return view('auth.login');
+        }
+    }
+
+    public function Login(Request $request)
+    {
+        $this->validateLogin($request);
+
+        if ($user = User::where('email', '=', $request->email)->exists()) {
+
+            $user = User::where('email', '=', $request->email)->first();
+
+        if ($user->use_sms_verify == true) {
+
+            $numberOfTry = 5;
+
+            if ($user = app('auth')->getProvider()->retrieveByCredentials($request->only('email', 'password'))) {
+                $token = Token::create([
+                    'user_id' => $user->id
+                ]);
+
+                if ($token->sendCode()) {
+                    session()->put("token_id", $token->id);
+                    session()->put("user_id", $user->id);
+                    session()->put("remember", $request->get('remember'));
+                    session()->put("number_of_try", $numberOfTry);
+
+                    return redirect("code");
+                }
+
+                $token->delete();// delete token because it can't be sent
+                return redirect('/login')->withErrors([
+                    "Unable to send verification code"
+                ]);
+            }
+
+            return redirect()->back()
+                ->withInputs()
+                ->withErrors([
+                    $this->username() => Lang::get('auth.failed'),
+                ]);
+
+        } else {
+            //dd($request);
+            $this->guard()->login($user, session()->get('remember', false));
+            return redirect('home');
+        }
+    }
+
+        return $this->sendFailedLoginResponse($request);
+
+    }
+
+    public function showCodeForm()
+    {
+
+
+        if (! session()->has("token_id")) {
+            return redirect("login");
+        }
+
+
+
+        return view("auth.code");
+    }
+
+    /**
+     * Store and verify user second factor.
+     * @param Request $request
+     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector
+     */
+    public function storeCodeForm(Request $request)
+    {
+
+
+        // throttle for too many attempts
+        if (! session()->has("token_id", "user_id")) {
+            return redirect("login");
+        }
+
+        $token = Token::find(session()->get("token_id"));
+        if (! $token ||
```

Comment:
do not use space between ! and expression
you can auto format your code using keyboard shortcut ctrl+alt+L (After highlighting the part of the text you want to format or for all file as well)